src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -496,17 +507,53 @@ public void testLinkedColumnNotDisplayedCase()
         customizeView.saveCustomView();
     }
 
+    @Test
+    public void testManualDatasetCategoryLink()
+    {
+        String categoryName = "CAT1";
+        createDatasetCategory(VISIT_BASED_STUDY, categoryName);
+        goToProjectHome();
+
+        log("Linking the sample type to preexisting dataset category");
+        linkToStudy(VISIT_BASED_STUDY, SAMPLE_TYPE1, 1, categoryName);
+
+        log("Linking the sample type to new dataset category");
+        goToProjectHome();
+        linkToStudy(DATE_BASED_STUDY, SAMPLE_TYPE2, 1, "CAT2");
```

Comment:
It looks like every time `linkToStudy` is called a call to `goToProjectHome` is done first. Why not pass in the project name and call `goToProjectHome` in `linkToStudy`?